sherpa/plot/__init__.py

```diff
@@ -72,23 +59,39 @@
 # TODO: why is this module globally changing the invalid mode of NumPy?
 _ = numpy.seterr(invalid='ignore')
 
-plot_opt = config.get('options', 'plot_pkg', fallback='dummy')
-plot_opt = [o.strip().lower() + '_backend' for o in plot_opt.split()]
-
 backend = None
 '''Currently active backend module for plotting.'''
 
+basicbackend = BasicBackend()
+'''This backend has all the defaults that are backend-independent.
+For all the plot classes in this module (e.g. FitPlot, JointPlot, ...)
+the default settings are set in code like `plot_prefs = xxx.get_plot_defaults()`
+where xxx is some backend. In the current design, this code is executed when this
+module imported and then the values are shared between all instances of a class.
+That allows one to change those default "globally" (i.e. for all instances of the class)
+but it means that the default values are based on the backend that is active when
+this module is imported. That depends on the sherpa.rc setting and thus `plot_prefs`
+might contain e.g. matplotlib specific defaults that are not applicable when other
+backends are used.
+Thus, we currently initialize them with the BasicBackend that has only the
+backend-indenpendent (i.e. those that works for any backend) defaults set.
+
+Note that this kind of indicates that the defaults don't have to be taken from a
+backend - they are static to the class and could live just there, but that
+will be addressed at a later stage.
+'''
```

Comment:
A lot of this text feels like it's more for a "design document" - e.g. comments about "current design" and "wil be addressed at a  later stage". I'm not sure they're particularly useful here.

Reply:
Yes, this is about design. I struggle with where to put notes like that. They should go somewhere and here in the code is the most likely place that a future developer finds them; I don't like to put it into a separate document in ClearCase that even I can't see. It could all be moved into the docs, or I could reference the github issue where I discussed this (and copied this text from). I think the answer is "a little bit of all of that". Long text needs to go into the docs (maybe on a page deep in the hierarchy called "Design choices for plotting backends", but I also want an explanation of why this backend is chosen to be initialized here. I don't think this is clearly right or wrong, it's a bit of personal preference how much of that to put in the code or in a separate document. I'll re-balance a bit in the iteration.